kafka/src/main/java/com/metamx/tranquility/kafka/writer/TranquilityEventWriter.java

```diff
@@ -90,31 +92,57 @@ public void send(byte[] message) throws InterruptedException
       );
     }
     catch (IOException e) {
-      if (++rejectedLogCount <= 5
+      final long rejectedLogCount = rejectedLogCounter.incrementAndGet();
+      if (rejectedLogCount <= 5
           || (rejectedLogCount <= 100 && rejectedLogCount % 10 == 0)
           || rejectedLogCount % 100 == 0) {
         log.debug(e, "%d message(s) failed to parse as JSON and were rejected", rejectedLogCount);
       }
-      if (dataSourceConfig.config().reportDropsAsExceptions()) {
-        Throwables.propagate(e);
+      if (dataSourceConfig.propertiesBasedConfig().reportDropsAsExceptions()) {
+        failedCounter.incrementAndGet();
```

Comment:
Perhaps failedCounter should be incremented regardless if we throw an exception or not.